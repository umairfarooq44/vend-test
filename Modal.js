import React from 'react';
import PropTypes from 'prop-types';
import { Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import { Icon } from 'antd';

export default class CustomModal extends React.Component{
    render() {
    const {
        textDirection,
        btnMargin,
        footerCancel,
        footerOkLoading,
        footerOk,
        footerOkClick,
        modalBody,
        isOpen,
        heading,
        close
    } = this.props;
    return(
        <Modal
            isOpen={isOpen}
            toggle={close}
            backdrop={'static'}
            className="dr-modal"
        >
            <ModalHeader>
                <Icon type="check-circle" />
                <h5>{heading}</h5>
            </ModalHeader>
            <ModalBody>
                <div className="row">
                    <div className={`col ml-auto ${textDirection}`}>
                    {modalBody}
                    </div>
                </div>
            </ModalBody>
            <ModalFooter>
            <button
                className={`btn btn-white border px-3 ${btnMargin}`}
                onClick={close}
            >
                {footerCancel}
            </button>
            <button
                disabled={footerOkLoading}
                className="btn btn-primary px-3"
                onClick={footerOkClick}
                style={{ width: '25%' }}
            >
                {footerOkLoading && (
                <span>
                    <i
                    className="fa fa-circle-o-notch fa-spin"
                    style={{ marginRight: '10px' }}
                    />{' '}
                </span>
                )}
                {footerOk}
            </button>{' '}
            </ModalFooter>
        </Modal>
    )
    }
}
CustomModal.propTypes = {
    textDirection: PropTypes.string,
    btnMargin: PropTypes.string,
    footerCancel: PropTypes.string,
    footerOkLoading: PropTypes.bool,
    footerOk: PropTypes.string,
    footerOkClick: PropTypes.func,
    modalBody: PropTypes.element,
    isOpen: PropTypes.bool,
    close: PropTypes.func,
    heading: PropTypes.string
};
CustomModal.defaultProps = {
    isOpen: false,
    footerOkLoading: false,
    heading: 'Are you sure?'
}