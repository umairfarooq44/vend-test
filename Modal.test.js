import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Modal from './Modal'

Enzyme.configure({adapter: new Adapter()});
it('should work fine', () => {
  const wrapper = shallow(<Modal />);
  expect(wrapper).toMatchSnapshot();
});

it('should have heading from default props', () => {
  const wrapper = shallow(<Modal />);
  expect(wrapper.find('h5').text()).toEqual('Are you sure?')
});

it('should have heading from props', () => {
  const wrapper = shallow(<Modal heading="my heading" />);
  expect(wrapper.find('h5').text()).toEqual('my heading')
});

it('should show loading ', () => {
  const wrapper = shallow(<Modal footerOkLoading={true} />);
  expect(wrapper.find('span')).toHaveLength(1)
});
it('should not show loading', () => {
  const wrapper = shallow(<Modal />);
  expect(wrapper.find('span')).toHaveLength(0);
});
it('should show cancel button text', () => {
  const wrapper = shallow(<Modal footerCancel="cancel" />);
  expect(wrapper.find('button').first().text()).toEqual('cancel');
});
it('should show ok button text', () => {
  const wrapper = shallow(<Modal footerOk="ok" />);
  expect(wrapper.find('button').last().text()).toEqual('ok');
});