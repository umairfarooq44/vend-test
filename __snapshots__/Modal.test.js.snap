// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work fine 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomModal
    footerOkLoading={false}
    heading="Are you sure?"
    isOpen={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoFocus": true,
      "backdrop": "static",
      "backdropTransition": Object {
        "mountOnEnter": true,
        "timeout": 150,
      },
      "centered": false,
      "children": Array [
        <ModalHeader
          charCode={215}
          closeAriaLabel="Close"
          tag="h5"
          wrapTag="div"
        >
          <Icon
            type="check-circle"
          />
          <h5>
            Are you sure?
          </h5>
        </ModalHeader>,
        <ModalBody
          tag="div"
        >
          <div
            className="row"
          >
            <div
              className="col ml-auto undefined"
            />
          </div>
        </ModalBody>,
        <ModalFooter
          tag="div"
        >
          <button
            className="btn btn-white border px-3 undefined"
          />
          <button
            className="btn btn-primary px-3"
            disabled={false}
            style={
              Object {
                "width": "25%",
              }
            }
          />
           
        </ModalFooter>,
      ],
      "className": "dr-modal",
      "fade": true,
      "isOpen": false,
      "keyboard": true,
      "modalTransition": Object {
        "timeout": 300,
      },
      "onClosed": [Function],
      "onOpened": [Function],
      "role": "dialog",
      "toggle": undefined,
      "zIndex": 1050,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "charCode": 215,
          "children": Array [
            <Icon
              type="check-circle"
            />,
            <h5>
              Are you sure?
            </h5>,
          ],
          "closeAriaLabel": "Close",
          "tag": "h5",
          "wrapTag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "type": "check-circle",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Are you sure?",
            },
            "ref": null,
            "rendered": "Are you sure?",
            "type": "h5",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col ml-auto undefined"
            />
          </div>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col ml-auto undefined"
            />,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "col ml-auto undefined",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <button
              className="btn btn-white border px-3 undefined"
            />,
            <button
              className="btn btn-primary px-3"
              disabled={false}
              style={
                Object {
                  "width": "25%",
                }
              }
            />,
            " ",
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "btn btn-white border px-3 undefined",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                false,
                undefined,
              ],
              "className": "btn btn-primary px-3",
              "disabled": false,
              "onClick": undefined,
              "style": Object {
                "width": "25%",
              },
            },
            "ref": null,
            "rendered": Array [
              false,
              undefined,
            ],
            "type": "button",
          },
          " ",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": "static",
        "backdropTransition": Object {
          "mountOnEnter": true,
          "timeout": 150,
        },
        "centered": false,
        "children": Array [
          <ModalHeader
            charCode={215}
            closeAriaLabel="Close"
            tag="h5"
            wrapTag="div"
          >
            <Icon
              type="check-circle"
            />
            <h5>
              Are you sure?
            </h5>
          </ModalHeader>,
          <ModalBody
            tag="div"
          >
            <div
              className="row"
            >
              <div
                className="col ml-auto undefined"
              />
            </div>
          </ModalBody>,
          <ModalFooter
            tag="div"
          >
            <button
              className="btn btn-white border px-3 undefined"
            />
            <button
              className="btn btn-primary px-3"
              disabled={false}
              style={
                Object {
                  "width": "25%",
                }
              }
            />
             
          </ModalFooter>,
        ],
        "className": "dr-modal",
        "fade": true,
        "isOpen": false,
        "keyboard": true,
        "modalTransition": Object {
          "timeout": 300,
        },
        "onClosed": [Function],
        "onOpened": [Function],
        "role": "dialog",
        "toggle": undefined,
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "charCode": 215,
            "children": Array [
              <Icon
                type="check-circle"
              />,
              <h5>
                Are you sure?
              </h5>,
            ],
            "closeAriaLabel": "Close",
            "tag": "h5",
            "wrapTag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "type": "check-circle",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Are you sure?",
              },
              "ref": null,
              "rendered": "Are you sure?",
              "type": "h5",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              className="row"
            >
              <div
                className="col ml-auto undefined"
              />
            </div>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col ml-auto undefined"
              />,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "col ml-auto undefined",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <button
                className="btn btn-white border px-3 undefined"
              />,
              <button
                className="btn btn-primary px-3"
                disabled={false}
                style={
                  Object {
                    "width": "25%",
                  }
                }
              />,
              " ",
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "btn btn-white border px-3 undefined",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  false,
                  undefined,
                ],
                "className": "btn btn-primary px-3",
                "disabled": false,
                "onClick": undefined,
                "style": Object {
                  "width": "25%",
                },
              },
              "ref": null,
              "rendered": Array [
                false,
                undefined,
              ],
              "type": "button",
            },
            " ",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
